packages/gestalt-datepicker/src/DateRange.js

```diff
@@ -0,0 +1,285 @@
+   */
+  startDateErrorMessage?: string | null,
+|};
+
+/**
+ * [DateRange](https://gestalt.pinterest.systems/web/daterange) enables users to preview and select a range of days by picking dates from a calendar or adding a text input.
+ * DateRange is distributed in its own package and must be installed separately.
+ *
+ * ![DateRange light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/DateRange.spec.mjs-snapshots/DateRange-chromium-darwin.png)
+ * ![DateRange dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/DateRange-dark.spec.mjs-snapshots/DateRange-dark-chromium-darwin.png)
+ */
+function DateRange({
+  endDateValue,
+  endDateErrorMessage,
+  localeData,
+  maxDate,
+  minDate,
+  onCancel,
+  onEndDateBlur,
+  onEndDateChange,
+  onEndDateError,
+  onEndDateFocus,
+  onStartDateBlur,
+  onStartDateChange,
+  onStartDateError,
+  onStartDateFocus,
+  onSubmit,
+  radioGroup,
+  startDateValue,
+  startDateErrorMessage,
+}: Props): Node {
+  const componentId = useId();
+  const deviceType = useDeviceType();
+  const isMobile = deviceType === 'mobile';
+
+  if (!startDateValue && endDateValue) {
+    onEndDateChange({ value: null });
+  }
+
+  const { applyText, cancelText } = useDefaultLabel('DateRange');
+
+  return (
+    <Box rounding={4} color="default" borderStyle="shadow" minHeight={425} display="inlineBlock">
+      <Flex>
+        {radioGroup &&
+        Children.only<Element<typeof RadioGroup>>(radioGroup).type.displayName === 'RadioGroup' &&
+        !isMobile ? (
+          <div className={borderStyles.borderRight}>
+            <Box paddingY={4} paddingX={6} width={216}>
+              {radioGroup}
+            </Box>
+          </div>
+        ) : null}
+        <Box>
+          <Flex alignItems="start" justifyContent="center" direction="column">
+            <div className={borderStyles.dateFieldSection}>
+              <Flex gap={3}>
+                <Box width={isMobile ? MOBILE_DATEFIELD_WIDTH : DATEFIELD_WIDTH}>
+                  <InternalDateField
+                    autoComplete="off"
+                    mobileEnterKeyHint="enter"
+                    id={`datefield-start-${componentId}`}
+                    localeData={localeData}
+                    onChange={({ value }) => {
+                      if (value?.getTime() || value === null) onStartDateChange({ value });
+                    }}
+                    onError={onStartDateError}
+                    onBlur={onStartDateBlur}
+                    onFocus={onStartDateFocus}
+                    value={startDateValue}
+                    minDate={minDate}
+                    maxDate={maxDate}
+                    errorMessage={startDateErrorMessage}
+                  />
+                </Box>
+                <Box display="flex" height="100%" alignItems="center">
```

Comment:
In an error state, this messes up

<img width="301" alt="image" src="https://github.com/pinterest/gestalt/assets/5509813/7913d233-6cbb-4c7e-9b8e-b0e2973f6b30">

Reply:
haha that's why I added a marginTop! @rlingineni

Reply:
DONE